flash/data/process.py

```diff
@@ -295,17 +324,34 @@ def __init__(
         self._predict_collate_in_worker_from_transform: Optional[bool] = None
         self._test_collate_in_worker_from_transform: Optional[bool] = None
 
-        self.train_transform = convert_to_modules(self._check_transforms(train_transform, RunningStage.TRAINING))
-        self.val_transform = convert_to_modules(self._check_transforms(val_transform, RunningStage.VALIDATING))
-        self.test_transform = convert_to_modules(self._check_transforms(test_transform, RunningStage.TESTING))
-        self.predict_transform = convert_to_modules(self._check_transforms(predict_transform, RunningStage.PREDICTING))
+        self._train_transform = self._check_transforms(train_transform, RunningStage.TRAINING)
+        self._val_transform = self._check_transforms(val_transform, RunningStage.VALIDATING)
+        self._test_transform = self._check_transforms(test_transform, RunningStage.TESTING)
+        self._predict_transform = self._check_transforms(predict_transform, RunningStage.PREDICTING)
 
-        if not hasattr(self, "_skip_mutual_check"):
-            self._skip_mutual_check = False
+        self.train_transform = convert_to_modules(self._train_transform)
+        self.val_transform = convert_to_modules(self._val_transform)
+        self.test_transform = convert_to_modules(self._test_transform)
+        self.predict_transform = convert_to_modules(self._predict_transform)
 
         self._callbacks: List[FlashCallback] = []
 
-    # todo (tchaton) Add a warning if a transform is provided, but the hook hasn't been overriden !
+    def _save_to_state_dict(self, destination, prefix, keep_vars):
+        preprocess_state_dict = self.save_state_dict()
+        if not isinstance(preprocess_state_dict, Dict):
+            raise MisconfigurationException("save_state_dict should return a dictionary")
+        preprocess_state_dict["_meta"] = {}
+        preprocess_state_dict["_meta"]["module"] = self.__module__
+        preprocess_state_dict["_meta"]["class_name"] = self.__class__.__name__
+        preprocess_state_dict["_meta"]["version"] = self.version()
+        try:
```

Comment:
parsing and setting any git attributes should be part of version, for example from my project:

https://github.com/jpuigcerver/PyLaia/blob/1b2e864247f1bfb8d95ac1910de9c52df71c017a/laia/__init__.py#L21-L31

Reply:
I thought about it, but I would prefer to leave this responsibility on the users. 
They should dump the version if the internal behaviour changed.
I am just saving the commit_sha so they can found this commit back on their history potentially.